CMakeLists.txt

```diff
@@ -132,24 +137,36 @@ target_sources(fortuno_wrapper PRIVATE
 
 # Testing program
 add_executable(run_kmeans run.f90)
-target_link_libraries(run_kmeans PRIVATE lib_kmeans OpenMP::OpenMP_Fortran ${MPI_Fortran_LIBRARIES} common_options) 
+target_link_libraries(run_kmeans PRIVATE lib_kmeans OpenMP::OpenMP_Fortran MPI::MPI_Fortran common_flags common_options) 
 
 # Library unit tests
 enable_testing()
 
 # Grids module
 add_executable(test_grids tests/test_grids_m.f90)
-target_link_libraries(test_grids PRIVATE lib_kmeans fortuno_wrapper)
-add_test(NAME test_generate_real_space_grid_1D COMMAND test_grids)
-add_test(NAME test_generate_real_space_grid_2D COMMAND test_grids)
-add_test(NAME test_generate_real_space_grid_3D COMMAND test_grids)
-add_test(NAME test_linspace COMMAND test_grids)
-add_test(NAME test_linspace_to_grid2d COMMAND test_grids)
-add_test(NAME test_generate_gaussian_2d COMMAND test_grids)
+target_link_libraries(test_grids PRIVATE lib_kmeans fortuno_wrapper common_flags common_options) 
```

Comment:
I knew there was a reason why each test binary was running all tests lol. Just had not got round to addressing it yet - thanks for the nudge